### ISSUE TITLE
Jetpack Sharing Buttons cause validation errors

### ISSUE THREAD

**Issue by jeremyclarke** [Automattic/facebook-instant-articles-wp#732]
# Steps required to reproduce the problem

1. Enable Jetpack
2. Enable "Add sharing buttons to your posts" setting in "Sharing" section of Jetpack settings
3. Use iA plugin on posts

# Expected Result

Both plugins are created by Automattic, and the iA plugin already has several fixes related to Jetpack, so:

* I expect everything to work and there to be no errors, so I can use both iA and Jetpack sharing at the same time

# Actual Result

* Transformation error: "No rules defined for `<div class="sharedaddy sd-sharing-enabled">` in the context of InstantArticle"

i.e. " This post will not be submitted to Instant Articles because the transformation raised some warnings." and " This post was transformed into an Instant Article with some warnings [Learn more | Transformer rule configuration | Toggle debug information]"
<img width="897" alt="screen shot 2017-07-19 at 15 08 10 pm" src="https://user-images.githubusercontent.com/139995/28384849-1cb3b9b4-6c94-11e7-9d84-d8ee9de76aae.png">

---

**Version Info**

* Plugin version: iA: 4.0.2 Jetpack: 5.1
* WordPress version: 4.8
* PHP version: 7.x

---

Is it possible this just hasn't been addressed yet? I can't find ANY references to the problem on the forums or here on GitHub, which seems outrageous considering how many people must be using both these plugins (and the fact that it would affect wordpress.com???)

I'm hoping something is wrong with my setup, but figured a ticket here was appropriate either way for future reference of others who might run into the problem.

**Comment by kraftbj** [comment]
x-posted from the JP issue. The AMP plugin removes the sharing buttons via https://github.com/Automattic/amp-wp/blob/master/jetpack-helper.php#L19-L21

**Comment by jeremyclarke** [comment]
Agree that something like that is exactly what I'd expect to be present for the iA plugin.

**Comment by jeremyclarke** [comment]
Here's the transformation code to make it work: 

`
{"rules":[{"class":"IgnoreRule","selector":".sharedaddy"}]}
`

Very simple, could be added without danger to the other WP rules.

**Comment by timjacobi** [comment]
Duplicate of #582

Please continue the discussion there.

**Comment by jeremyclarke** [comment]
Um I have to dissagree. This ticket is about a specific Automattic plugin's feature, and contains a specific solution. That one is a mess from someone who doesn't understand their own problem. 

Why would you close this and keep that open? Someone has to fix Jetpack+Instant Articles and they've already punted it on the Jetpack side in favor of this ticket.

**Comment by timjacobi** [comment]
The mark-up you have posted suggest that fixing the more general `div` problem will transitively fix this problem too. Do you expect anything else to break once this is fixed? 

I did mention in #582 that we will verify the referenced issues including this one are fixed once the general fix is in.

**Comment by jeremyclarke** [comment]
My fix here isn't at all a generic div fix though, so that's the misunderstanding. 

For iA, each class of div needs different handling. 

Some of them should be ignored like sharedaddy, because they are irrelevant to the post content and useless in the context of an iA reader. 

Others need to be converted into one of the iA formats (e.g. if you had a pull quote plugin it's div.pull-quote would become an iA pullquote) so that the important content inside gets rendered meaningfully in the iA version.

Other classes of div need to simply be left alone, and cause iA validation errors, because the content is important to the story and can't be rendered in iA. 

This plugin doesn't have to solve every div example, but it should take responsibility for deciding which it will handle and which it won't. Either way, clumping all div related problems together gets us nowhere.

**Comment by jeremyclarke** [comment]
The meta problem that needs to be solved is how miserable it is for plugin authors to add iA transformation support for their output, which dumps it in the laps of users who certainly can't be expected to write JSON by hand. That's a whole other ticket if you want to create it, then punt the use of that API to Jetpack for this ticket once the other one is done.

**Comment by jeremyclarke** [comment]
This really shouldn't be closed. The reason is this:

JETPACK SHARING TOOLS BREAKS THE AUTOMATTIC INSTANT ARTICLES PLUGIN

If that is okay then I don't know what to say.
